packages/contracts/contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol

```diff
@@ -1326,16 +1335,13 @@ contract OVM_ExecutionManager is iOVM_ExecutionManager, Lib_AddressResolver {
 
     // Check whether the account has been changed before and mark it as changed if not. We need
     // this because "nuisance gas" only applies to the first time that an account is changed.
-    bool _wasAccountAlreadyChanged =
-      ovmStateManager.testAndSetAccountChanged(_address);
-
-    // If we hadn't already loaded the account, then we'll need to charge "nuisance gas" based
+    // If we haven't already changed the account, then we'll need to charge "nuisance gas" based
     // on the size of the contract code.
+    bool _wasAccountAlreadyChanged = ovmStateManager.testAndSetAccountChanged(_address);
     if (_wasAccountAlreadyChanged == false) {
       ovmStateManager.incrementTotalUncommittedAccounts();
       _useNuisanceGas(
-        (Lib_EthUtils.getCodeSize(_getAccountEthAddress(_address)) *
-          NUISANCE_GAS_PER_CONTRACT_BYTE) + MIN_NUISANCE_GAS_PER_CONTRACT
+        (Lib_EthUtils.getCodeSize(_getAccountEthAddress(_address)) * NUISANCE_GAS_PER_CONTRACT_BYTE)
```

Comment:
Yeah I guess we can remove the fixed cost here if we shift it to the fixed cost of the account loading step.

Reply:
That was my thinking, yes. But now I'm second guessing that. 
I think the purpose of `MIN_NUISANCE_GAS_PER_CONTRACT` is to cover the cost of `proveContractState` and `commitContractState` even for a very small contract. 

Imagine an account starts empty, and ends with a single byte of code. In that case you'd want to ensure that `MIN_NUISANCE_GAS_PER_CONTRACT` is charged both times.